python/src/array.cpp

```diff
@@ -16,13 +16,6 @@
 namespace py = pybind11;
 using namespace py::literals;
 
-enum PyScalarT {
-  pybool = 0,
-  pyint = 1,
-  pyfloat = 2,
-  pycomplex = 3,
-};
-
```

Comment:
Replace it with `Dtype` since we'd have to deal with all possible mlx dtypes when create from list of mlx.arrays. Not just python primitive types.